gef-ui/server/controllers/amendments/facility-end-date/post-facility-value.test.ts

```diff
@@ -0,0 +1,274 @@
+  });
+
+  it('should call getApplication with the correct dealId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await postFacilityEndDate(req, res);
+
+    // Assert
+    expect(getApplicationMock).toHaveBeenCalledTimes(1);
+    expect(getApplicationMock).toHaveBeenCalledWith({ dealId, userToken: req.session.userToken });
+  });
+
+  it('should call getFacility with the correct facilityId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await postFacilityEndDate(req, res);
+
+    // Assert
+    expect(getFacilityMock).toHaveBeenCalledTimes(1);
+    expect(getFacilityMock).toHaveBeenCalledWith({ facilityId, userToken: req.session.userToken });
+  });
+
+  it('should not call updateAmendment if the facilityEndDate is invalid', async () => {
+    // Arrange
+    const facilityEndDateDayMonthYear = {
+      day: '100',
+      month: '100',
+      year: '100',
+    };
+    const { req, res } = getHttpMocks(facilityEndDateDayMonthYear);
+
+    // Act
+    await postFacilityEndDate(req, res);
+
+    // Assert
+    expect(updateAmendmentMock).toHaveBeenCalledTimes(0);
+  });
+
+  it('should render the page with validation errors if facilityEndDate is invalid', async () => {
+    // Arrange
+    const facilityEndDateDayMonthYear = {
+      day: '100',
+      month: '100',
+      year: '100',
+    };
+    const { req, res } = getHttpMocks(facilityEndDateDayMonthYear);
+
+    // Act
+    await postFacilityEndDate(req, res);
+
+    // Assert
+    const expectedRenderData: FacilityEndDateViewModel = {
+      exporterName: mockDeal.exporter.companyName,
+      cancelUrl: `/gef/application-details/${dealId}/facilities/${facilityId}/amendments/${amendmentId}/cancel`,
+      previousPage,
+      errors: validationErrorHandler(
+        (validateAndParseFacilityEndDate(facilityEndDateDayMonthYear, new Date(MOCK_ISSUED_FACILITY.details.coverStartDate)) as { errors: ValidationError[] })
+          .errors,
+      ),
+      facilityEndDate: facilityEndDateDayMonthYear,
+    };
+
+    expect(res._getStatusCode()).toEqual(HttpStatusCode.Ok);
+    expect(res._getRenderView()).toEqual('partials/amendments/facility-end-date.njk');
+    expect(res._getRenderData()).toEqual(expectedRenderData);
+  });
+
+  it('should call updateAmendment if the facilityEndDate is valid', async () => {
+    // Arrange
+    const facilityEndDateDayMonthYear = { day: format(today, 'd'), month: format(today, 'M'), year: format(today, 'yyyy') };
+    const { req, res } = getHttpMocks(facilityEndDateDayMonthYear);
+
+    // Act
+    await postFacilityEndDate(req, res);
+
+    // Assert
+    expect(updateAmendmentMock).toHaveBeenCalledTimes(1);
+    expect(updateAmendmentMock).toHaveBeenCalledWith({ facilityId, amendmentId, update: { facilityEndDate: today }, userToken });
+  });
+
+  it('should not call console.error if the facilityEndDate is valid', async () => {
```

Comment:
I don't see any date minimum and maximum bounds, I suppose there is no business logic to stipulate this?